lib/field/use-field-like-sync.ts

```diff
@@ -9,74 +9,77 @@ export interface UseFieldLikeSyncProps<
   F,
   TT extends FieldInstance<T, F> | FieldArrayInstance<T, F>
 > {
-  mutableRef: MutableRefObject<TT>;
+  fieldInstanceRef: MutableRefObject<TT>;
   props: TT["props"];
   value: TT["value"];
   errors: TT["errors"];
   isDirty: TT["isDirty"];
   isValid: TT["isValid"];
   isTouched: TT["isTouched"];
   isValidating: TT["isValidating"];
+  preserveValue?: boolean;
 }
 
 export const useFieldLikeSync = <
   T,
   F,
   TT extends FieldInstance<T, F> | FieldArrayInstance<T, F>
 >({
-  mutableRef,
+  fieldInstanceRef,
   props,
   value,
   errors,
   isDirty,
   isValid,
   isTouched,
   isValidating,
+  preserveValue,
 }: UseFieldLikeSyncProps<T, F, TT>) => {
   const formContext = useFormContext<F>();
 
   /**
    * Add mutable ref to formFieldsRef
    */
   useIsomorphicLayoutEffect(() => {
-    mutableRef.current.props = props;
-    const newMutable = mutableRef.current;
-    formContext.formFieldsRef.current.push(newMutable);
+    fieldInstanceRef.current.props = props;
+    const fieldInstance = fieldInstanceRef.current;
+    const formFields = formContext.formFieldsRef.current;
+    const found = formFields.find((field) => field === fieldInstance);
+    if (!found) formFields.push(fieldInstance);
```

Comment:
Oh! I just realized what this is for - this is so that we preserve the internal values of the fields if/when they're re-rendered. This is a GREAT insight that I wouldn't've considered. Do we have tests for this specific behavior?

Reply:
I'm not sure if I understood well but I added this because if `preserveValue` is true the cleanup function was not being executed so it was happening to have duplicate fields in `formFieldsRef`

Reply:
Yeah, this makes perfect sense to me. Good catch